ouroboros/src/validator/mod.rs

```diff
@@ -0,0 +1,7 @@
+/// Generic trait for validating any type of data. Designed to be used across threads so validations
+/// can be done in parallel. Validators are expected to never result in an error, but instead panic
+/// if the validation is unable to be completed and return a true or false. This indicates a serious
+/// bug in the code. Validators should handle all error cases internally.
+pub trait Validator: Send + Sync {
+    fn validate(&self) -> bool;
+}
```

Comment:
I am unsure to understand the intent based on that comment? If validators cannot error, and are considered buggy if they return `false`; what are we validating here exactly?

Reply:
The comment can be improved. Basically, validators should always return true (valid), or false (invalid). All validation errors should be handled internally and return a false.

I saw no point in propagating error messages out of the validator. There's never any special behavior a caller needs to do based on certain types of errors. Things are either valid or they aren't.

Reply:
Fixed comment.